website/docs/cli/workspaces/index.mdx

```diff
@@ -7,72 +7,99 @@ description: >-
-- Use [the `terraform workspace select` command](/cli/commands/workspace/select)
-  to change the currently selected workspace.
-- Use the [`terraform workspace list`](/cli/commands/workspace/list),
-  [`terraform workspace new`](/cli/commands/workspace/new), and
-  [`terraform workspace delete`](/cli/commands/workspace/delete) commands
-  to manage the available workspaces in the current working directory.
-
--> **Note:** Terraform Cloud and Terraform CLI both have features called
-"workspaces," but they're slightly different. Terraform Cloud's workspaces
-behave more like completely separate working directories.
-
-## The Purpose of Workspaces
-
-Since most of the resources you can manage with Terraform don't include a unique
-name as part of their configuration, it's common to use the same Terraform
-configuration to provision multiple groups of similar resources.
-
-Terraform relies on [state](/language/state) to associate resources with
-real-world objects, so if you run the same configuration multiple times with
-completely separate state data, Terraform can manage many non-overlapping groups
-of resources. In some cases you'll want to change
-[variable values](/language/values/variables) for these different
-resource collections (like when specifying differences between staging and
-production deployments), and in other cases you might just want many instances
-of a particular infrastructure pattern.
-
-The simplest way to maintain multiple instances of a configuration with
-completely separate state data is to use multiple
-[working directories](/cli/init) (with different
-[backend](/language/settings/backends/configuration) configurations per directory, if you
-aren't using the default `local` backend).
-
-However, this isn't always the most _convenient_ way to handle separate states.
-Terraform installs a separate cache of plugins and modules for each working
-directory, so maintaining multiple directories can waste bandwidth and disk
-space. You must also update your configuration code from version control
-separately for each directory, reinitialize each directory separately when
-changing the configuration, etc.
-
-Workspaces allow you to use the same working copy of your configuration and the
-same plugin and module caches, while still keeping separate states for each
-collection of resources you manage.
+Workspaces in the Terraform CLI refer to separate instances of [state data](/language/state) inside the same working directory.
+
+Terraform relies on state to associate resources with real-world objects. When you run the same configuration multiple times with separate state data, Terraform can manage multiple sets of non-overlapping resources.
+
+Workspaces let you create many instances of the same infrastructure or inject different [variable values](/language/values/variables) for each instance. For example, you may want to quickly create infrastructure to test changes without affecting the production deployment.
```

Comment:
I think the "let you ... inject different variable values" statement here is a bit tricky...

It reads to me like it's saying that different variable values per workspace are a built-in feature of workspaces, but that isn't true. Instead, I think it's perhaps more true to say something like this:

> Workspaces let you create many instances of the same infrastructure. You can describe small differences between those instances by setting different [variable values](/language/values/variables) for each instance.

I'm not sure this is totally clear either, so I will say the same thing in more words just in case: _If_ you choose to use multiple workspaces then you will typically _also_ need to manually specify different input variables for each distinct workspace, because otherwise they'll all be describing exactly the same infrastructure and will inevitably collide with one another. So injecting different input variables is something you start needing to do as a consequence of using workspaces, not a _benefit_ of workspaces. Hopefully you can see what I mean with all of this incoherent jibber-jabber. :upside_down_face:

Reply:
ahhh okay this is very helpful! I'm going to push up a fix for this to make it clearer - thank you!